### PR TITLE
Build Edge commit and installer and upload to S3

### PR DESCRIPTION
**Pull request by achilleas-k**
Counterpart to https://github.com/osbuild/osbuild-composer/pull/1439

This PR adds a new upload type labelled `aws.s3` for uploading artifacts to an S3 bucket.

## EDIT

There are some things I'd like to include before marking as ready:
- [x] OSTree parameters
- [x] Edge installer: COMPOSER-926
- [x] Tests

The edge installer would receive the same treatment as the edge commit, but it will need to pull the edge commit payload to embed it.

### CODE REVIEW

**Review by teg — COMMENTED** [review]
This is great! Thanks!

File: distributions/rhel-85.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "module_platform_id": "platform:el8",
+  "distribution": {
+    "name": "rhel-85",
+    "description": "Red Hat Enterprise Linux (RHEL) 8"
+  },
+  "x86_64": {
+    "image_types": [ "ami", "vhd", "edge-commit" ],
```
Comment (teg) [inline distributions/rhel-85.json:8]: I think this should just be added as part of 8.4 instead, no?

**Review by teg — CHANGES_REQUESTED** [review]
I think if we change the patch to add 8.5 support to instead add edge support to 8.4, this can be landed (as soon as the composer counter-part).

File: distributions/rhel-85.json (hunk 1 above)
Comment (Gundersanne) [reply distributions/rhel-85.json:8]: if we just want edge support in GA then we can just add it to the current rhel-8 file

Comment (teg) [reply distributions/rhel-85.json:8]: GA?

**Comment by achilleas-k** [comment]
Added `rhel-edge-commit` to the RHEL 8.4 distribution config.

**Review by teg — DISMISSED** [review]
This is really great! Could we hook up some basic test too before merging it?

**Comment by achilleas-k** [comment]
I've added basic unit tests.  Tested building a `rhel-edge-installer` locally and no further changes are needed other than including it in the distribution file.

**Comment by teg** [comment]
I think a fixup commit is still left over. @Gundersanne do we do full build/upload tests for the other image types yet, do we need the same here?

**Comment by achilleas-k** [comment]
oops, sorry for the fixup commit leftover.  Just noticed it now while adding the installer to the distro config.
Fixed.

**Comment by Gundersanne** [comment]
> I think a fixup commit is still left over. @Gundersanne do we do full build/upload tests for the other image types yet, do we need the same here?

Yea all the images are built/uploaded (rhel only though). I'm ok with it being added in a followup however.

**Review by Gundersanne — APPROVED** [review]
This looks fine. We probably want to include an integration test? But I'm fine with doing that in a followup.

**Comment by Gundersanne** [comment]
Added a blocked label for osbuild/osbuild-composer#1439 ; as soon as that's in we can redeploy composer and merge this!

**Comment by achilleas-k** [comment]
Moved the edge image types from `rhel-8.json` to `rhel-84.json` (see #202)

**Comment by achilleas-k** [comment]
Changed the location of the ostree options in the request payload.  See also https://github.com/osbuild/osbuild-composer/pull/1439#issuecomment-863157741.

**Comment by teg** [comment]
I think we are very, very close to having the composer PR in, this will need a rebase though.

**Comment by achilleas-k** [comment]
Unblocked and rebased.

**Review by teg — APPROVED** [review]
Thanks!

**Pull request merged**